### PR TITLE
[iOS] Use updated APIs for the button with images

### LINKED ISSUE #9734: Button with image not scaling to HeightRequest on iOS and Android
### Description

Button with an image fails to correctly scale the source image to the size of the button this results in the full size image being displayed in Android and a skewed image in iOS

iOS
Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-08-29 at 17 43 58

Android
Screenshot_1661759181

WinUI
Screenshot 2022-08-29 180009

### Steps to Reproduce

Create a new example Maui App

Add the following line of xaml to MainPage.xaml

```
<Button
                ImageSource="dotnet_bot.png"
                Text="Im a happy button"
                HeightRequest="40"
                WidthRequest="200"/>
```

Run the solution, works correctly in WinUI and results in the full sized image next to the button in Android and the image skewed and distorted but with the correct height in iOS 

I expect this to correctly render the image in the button to the left of the text in the button using the heightrequest to scale down to the requested height maintaining the images aspect ratio, see WinUI example above where the image and button are rendered correctly

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS 15.5, Android 33

### Did you find any workaround?

Reduce the image size manually to be the correct size desired on a device

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by tj-devel709**
### Description of Change

~~Cherry pick iOS only changes from https://github.com/dotnet/maui/pull/19834~~
This PR updates a deprecated API in iOS to use the newer UIButton.Configuration API. Resizing is not really an option anymore with the new API and requires creating a new image each time. For now, let's update to the new API.

### Issues Fixed

Makes progress towards #9734
Makes progress towards #21394

### CODE REVIEW

**Comment by jsuarezruiz** [comment]
> Bringing over [this comment](https://github.com/dotnet/maui/pull/19834#issuecomment-1973876085):
> 
> It appears that there is an issue on iOS with an ImageButton inside a scrollview. The image does not resize with the width request as android and Windows does. Check out the bottom imagebutton in each of the screenshots:
> 
> iOS: image
> 
> Android: image
> 
> Windows: image
> 
> This is the relevant xaml:
> 
> ```
> <ScrollView HeightRequest="100" WidthRequest="50">
>    <ImageButton Source="dotnet_bot.png" />
> </ScrollView>
> ```

Could we move `ResizeImageSource `extension method to the ButtonExtensions and then, use the same on the `ImageButtonImageSourcePartSetter `from ButtonHandler and ImageButtonHandler?

File: src/Controls/samples/Controls.Sample.UITests/Issues/Issue18242.xaml.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 namespace Maui.Controls.Sample.Issues
 {
 	[XamlCompilation(XamlCompilationOptions.Compile)]
-	[Issue(IssueTracker.Github, 18242, "Button ImageSource not Scaling as expected", PlatformAffected.UWP)]
+	[Issue(IssueTracker.Github, 18242, "Button ImageSource not Scaling as expected", PlatformAffected.All)]
```
Comment (jsuarezruiz) [inline src/Controls/samples/Controls.Sample.UITests/Issues/Issue18242.xaml.cs:7]: Could we include here a sample using a different ContentLayout (for example, Top)?
Could we include a case using dynamic size?. We can include a button, and tapping it just increase at runtime the height using a random value.

**Review by jsuarezruiz — COMMENTED** [review]
image
Is the first Button behavior an expected result?

**Comment by tj-devel709** [comment]
Since this PR is not addressing resizing, removed the existing UITest for iOS that was testing resizing and instead added a UITest with a screenshot test to make sure the different image placements work with the button. Will wait for the UITest to produce the diffs!

File: src/Controls/tests/UITests/Tests/Issues/Issue21394.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		{
+			this.Ignore(TestDevice.iOS, () => {
+                if (!OperatingSystem.IsIOSVersionAtLeast(15))
+                    return true;
+                return false;
+            }, "iOS 14 and below sizes button images differently.");
+
+			App.WaitForElement("WaitForStubControl");
+
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/UITests/Tests/Issues/Issue21394.cs:26]: The snapshots for Android and Windows are pending.
image

**Comment by tj-devel709** [comment]
The windows screenshot is looking a little funky. Testing on windows

**Comment by PureWeen** [comment]
Possibly related
https://github.com/dotnet/maui/issues/21488

**Pull request closed without merging**